layouts/partials/helpers/fragments.html

```diff
@@ -14,15 +14,9 @@
   check for presence of local fragments since multiple files can use a single
   fragment.
   */}}
-{{- $local_fragments := $real_page.Resources.ByType "page" -}}
-
-{{- $page_scratch.Set "fragments" (slice $page) -}}
-{{- $page_scratch.Set "local_fragments_dirs" (slice) -}}
-{{- range $local_fragments -}}
-  {{- $page_scratch.Add "fragments" (slice .) -}}
-  {{- $page_scratch.Add "local_fragments_dirs" .Dir -}}
-{{- end -}}
 
+{{/* We try to find every global directory from the parent directories and the
+  local fragments */}}
 {{- $page_scratch.Set "sections" (slice "/") -}}
 {{- $sections := findRE "[^\\/]+" $real_page.CurrentSection.Dir -}}
```

Comment:
Considering other devs. I think it should be documented why we use findRE to find the sections.